File: packages/flutter_amap/ios/Classes/AMapSdkApi.swift
import Flutter
// 地图独立集成使用 MAMapKit；地图与导航联合集成时复用 AMapNaviKit 内含的地图能力。
#if canImport(MAMapKit)
import MAMapKit
#elseif canImport(AMapNaviKit)
import AMapNaviKit
#else
#error("Neither MAMapKit nor AMapNaviKit is available. Check flutter_amap's iOS SDK configuration.")
#endif

class _AMapSdkApi: NSObject {
  static func setup(registrar: FlutterPluginRegistrar) {
    let initializerChannel = FlutterMethodChannel(name: "plugins.flutter.dev/amap_initializer", binaryMessenger: registrar.messenger())
    initializerChannel.setMethodCallHandler({ (call: FlutterMethodCall, result: FlutterResult) -> Void in
      if(call.method == "agreePrivacy") {
        let arguments = call.arguments as! Dictionary<String, AnyObject>
        let agree = arguments["agree"] as! Bool
        _AMapSdkApi.agreePrivacy(agreePrivacy: agree)
        result(nil)
      } else if(call.method == "setApiKey") {
        let arguments = call.arguments as! Dictionary<String, AnyObject>
        let apiKey = arguments["iosKey"] as! String
        _AMapSdkApi.setApiKey(apiKey: apiKey)
        result(nil)
      }
    })
  }

  static func agreePrivacy(agreePrivacy: Bool) {
    MAMapView.updatePrivacyShow(AMapPrivacyShowStatus.init(agreePrivacy), privacyInfo: AMapPrivacyInfoStatus.init(agreePrivacy))
    MAMapView.updatePrivacyAgree(AMapPrivacyAgreeStatus.init(agreePrivacy))
  }

  static func setApiKey(apiKey: String) {
    AMapServices.shared().apiKey = apiKey
  }
}

File: packages/flutter_amap/android/src/main/kotlin/com/morbit/amap_flutter/AMapSdkApi.kt
package com.morbit.amap_flutter

import android.content.Context
import com.amap.api.maps.MapsInitializer
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class AMapSdkApi {
  companion object {
    fun setup(binding: FlutterPluginBinding) {
      val initializerChannel = MethodChannel(binding.binaryMessenger, "plugins.flutter.dev/amap_initializer")
      initializerChannel.setMethodCallHandler { call: MethodCall, result: MethodChannel.Result ->
        when (call.method) {
          "agreePrivacy" -> {
            val agree = call.argument<Boolean>("agree")!!
            agreePrivacy(binding.applicationContext, agree)
            result.success(null)
          }
          "setApiKey" -> {
            val apiKey = call.argument<String>("androidKey")!!
            setApiKey(apiKey)
            result.success(null)
          }
          "getSdkVersion" -> result.success(MapsInitializer.getVersion())
          else -> result.notImplemented()
        }
      }
    }

    private fun agreePrivacy(context: Context, agreePrivacy: Boolean) {
      MapsInitializer.updatePrivacyShow(context, agreePrivacy, agreePrivacy)
      MapsInitializer.updatePrivacyAgree(context, agreePrivacy)
    }

    private fun setApiKey(apiKey: String) {
      MapsInitializer.setApiKey(apiKey)
    }
  }
}
